src/utils.jl

```diff
@@ -2,24 +2,14 @@
 # Licensed under the MIT License. See LICENCE in the project root.
 # ------------------------------------------------------------------
 
-const GPU = nothing
-
-function get_imfilter_impl(GPU)
-  if GPU ≠ nothing
-    imfilter_gpu
-  else
-    imfilter_cpu
-  end
-end
-
-function convdist(img::AbstractArray, kern::AbstractArray;
-                  weights::AbstractArray=fill(1.0, size(kern)))
-  imfilter_impl = get_imfilter_impl(GPU)
+function convdist(resource::AbstractResource{R},
+                  img::AbstractArray, kern::AbstractArray;
+                  weights::AbstractArray=fill(1.0, size(kern))) where {R}
```

Comment:
```suggestion
function convdist(resource, img, kern; weight=fill(1.0, size(kern)))
```
Cleanup type annotations as they don't help with dispatch.